pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -74,21 +98,25 @@ func (s *Service) Handshake(stream p2p.Stream) (i *Info, err error) {
 		return nil, fmt.Errorf("read synack message: %w", err)
 	}
 
-	address := swarm.NewAddress(resp.Syn.Address)
-	if resp.Syn.NetworkID != s.networkID {
-		return nil, ErrNetworkIDIncompatible
+	if err := s.checkAck(resp.Ack); err != nil {
+		return nil, err
+	}
+
+	if err := s.checkSyn(resp.Syn); err != nil {
+		return nil, err
 	}
 
 	if err := w.WriteMsgWithTimeout(messageTimeout, &pb.Ack{
-		Address: resp.Syn.Address,
+		BzzAddress: resp.Syn.BzzAddress,
 	}); err != nil {
 		return nil, fmt.Errorf("write ack message: %w", err)
 	}
 
-	s.logger.Tracef("handshake finished for peer %s", address)
+	s.logger.Tracef("handshake finished for peer %s", resp.Syn.BzzAddress.Overlay)
```

Comment:
Overlay is []byte and I think that it will not be nicely printed with %s. Maybe to create an overlay variable and to use it here and in the returned Info?

Reply:
Sure, make sense!